lib/src/helpers/http.dart

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+import 'dart:typed_data';
+
+import '../../web.dart';
+import 'events/events.dart' show XHRGetters;
+
+/// A helper used to make it easier to operate over [XMLHttpRequest]s.
+///
+/// The logic here was copied from `dart:html` to help bridge a functionality
+/// gap missing in `package:web`.
+///
+/// HttpRequest can be used to obtain data from HTTP and FTP protocols,
+/// and is useful for AJAX-style page updates.
+///
+/// The simplest way to get the contents of a text file, such as a
+/// JSON-formatted file, is with [getString].
+/// For example, the following code gets the contents of a JSON file
+/// and prints its length:
+///
+///     var path = 'myData.json';
+///     HttpRequest.getString(path).then((String fileContents) {
+///       print(fileContents.length);
+///     }).catchError((error) {
+///       print(error.toString());
+///     });
+///
+/// ## Fetching data from other servers
+///
+/// For security reasons, browsers impose restrictions on requests
+/// made by embedded apps.
+/// With the default behavior of this class,
+/// the code making the request must be served from the same origin
+/// (domain name, port, and application layer protocol)
+/// as the requested resource.
+/// In the example above, the myData.json file must be co-located with the
+/// app that uses it.
+///
+/// ## Other resources
+///
+/// * [Fetch data dynamically](https://dart.dev/tutorials/web/fetch-data/),
+/// a tutorial shows how to load data from a static file or from a server.
+/// * [Dart article on using HttpRequests](https://dart.dev/guides/libraries/library-tour#using-http-resources-with-httprequest)
+/// * [JS XMLHttpRequest](https://developer.mozilla.org/en-US/docs/DOM/XMLHttpRequest)
+/// * [Using XMLHttpRequest](https://developer.mozilla.org/en-US/docs/DOM/XMLHttpRequest/Using_XMLHttpRequest)
+
+class HttpRequest {
```

Comment:
Should have a TODO Here to provide a `pkg:http` wrapper!

Reply:
I don't follow - shouldn't that be in pkg:http instead? (replace their dart:html import by a package:web import?)

Reply:
Eventually. I guess we should keep this simple. We should do that migration as soon as we can, though...